ostree-ext/src/chunking.rs

```diff
@@ -323,49 +351,72 @@ impl Chunking {
             .try_into()
             .unwrap();
 
-        // TODO: Compute bin packing in a better way
-        let start = Instant::now();
-        let packing = basic_packing(
-            sizes,
-            NonZeroU32::new(self.max).unwrap(),
-            prior_build_metadata,
-        )?;
-        let duration = start.elapsed();
-        tracing::debug!("Time elapsed in packing: {:#?}", duration);
-
-        for bin in packing.into_iter() {
-            let name = match bin.len() {
-                0 => Cow::Borrowed("Reserved for new packages"),
-                1 => {
-                    let first = bin[0];
-                    let first_name = &*first.meta.identifier;
-                    Cow::Borrowed(first_name)
-                }
-                2..=5 => {
-                    let first = bin[0];
-                    let first_name = &*first.meta.identifier;
-                    let r = bin.iter().map(|v| &*v.meta.identifier).skip(1).fold(
-                        String::from(first_name),
-                        |mut acc, v| {
-                            write!(acc, " and {}", v).unwrap();
-                            acc
-                        },
-                    );
-                    Cow::Owned(r)
-                }
-                n => Cow::Owned(format!("{n} components")),
-            };
-            let mut chunk = Chunk::new(&name);
-            chunk.packages = bin.iter().map(|v| String::from(&*v.meta.name)).collect();
-            for szmeta in bin {
-                for &obj in rmap.get(&szmeta.meta.identifier).unwrap() {
-                    self.remainder.move_obj(&mut chunk, obj.as_str());
+        // Filter out exclusive components for regular packing
+        let regular_sizes: Vec<ObjectSourceMetaSized> = sizes
+            .iter()
+            .filter(|component| {
+                !processed_specific_components.contains(&*component.meta.identifier)
+            })
+            .map(|component| ObjectSourceMetaSized {
+                meta: ObjectSourceMeta {
+                    identifier: Rc::clone(&component.meta.identifier),
+                    name: Rc::clone(&component.meta.name),
+                    srcid: Rc::clone(&component.meta.srcid),
+                    change_time_offset: component.meta.change_time_offset,
+                    change_frequency: component.meta.change_frequency,
+                },
+                size: component.size,
+            })
+            .collect();
+
+        // Process regular components with bin packing if we have remaining layers
+        if !regular_sizes.is_empty() && self.remaining() > 0 {
```

Comment:
Why the explicit check for `if !regular_sizes.is_empty()` - if I understand correctly what would happen in the case where *all* content is explicitly specified is that we'd still end up with the "resreved for new packages" one, and that should be it?

I think we could special case that one. 

(The "reserved for new packages" thing I think never worked out...but that's a different issue)